File: UploadfileS3.py
import os
from flask import Flask, request, jsonify, redirect, flash
import boto3
from werkzeug.utils import secure_filename
from dotenv import load_dotenv
from flask_cors import CORS
from werkzeug.datastructures import FileStorage

# Load the environment variables from the .env file
load_dotenv()
print("AWS_ACCESS_KEY_ID:", os.environ.get("AWS_ACCESS_KEY_ID"))
print("S3_BUCKET_NAME:", os.environ.get("S3_BUCKET_NAME"))

# Now your environment variables are accessible via os.environ.get()
AWS_ACCESS_KEY_ID = os.environ.get("AWS_ACCESS_KEY_ID")
AWS_SECRET_ACCESS_KEY = os.environ.get("AWS_SECRET_ACCESS_KEY")
S3_BUCKET_NAME = os.environ.get("S3_BUCKET_NAME")
S3_REGION = os.environ.get("S3_REGION", "us-east-1")

app = Flask(__name__)
# Enable CORS and allow credentials
CORS(app, supports_credentials=True)
app.secret_key = "your_secret_key"  # Replace with a random secret for production

# Initialize the boto3 S3 client
s3_client = boto3.client(
    "s3",
    aws_access_key_id=AWS_ACCESS_KEY_ID,
    aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
    region_name=S3_REGION
)

def upload_file_to_s3(file, bucket_name, filename, content_type):
    """
    Uploads a file object to S3 using boto3.
    Returns the public URL of the uploaded file.
    """
    try:
        # Upload the file to S3
        s3_client.upload_fileobj(
            file,
            bucket_name,
            filename,
            ExtraArgs={
                "ContentType": content_type,
                "ACL": "public-read"
            }
        )
        
        # Construct the public URL
        file_url = f"https://{bucket_name}.s3.{S3_REGION}.amazonaws.com/{filename}"
        return file_url
    except Exception as e:
        print(f"Error uploading file to S3: {str(e)}")
        return None

@app.route("/upload", methods=["POST"])
def upload_file():
    try:
        # Check if file is present in the request
        if 'file' not in request.files:
            return jsonify({"error": "No file part"}), 400
            
        file = request.files['file']
        if file.filename == '':
            return jsonify({"error": "No selected file"}), 400
            
        # Get additional parameters
        filename = request.form.get('fileName', secure_filename(file.filename))
        content_type = request.form.get('contentType', 'application/octet-stream')
        
        # Upload the file to S3
        file_url = upload_file_to_s3(file, S3_BUCKET_NAME, filename, content_type)
        
        if file_url:
            return jsonify({"url": file_url})
        else:
            return jsonify({"error": "Failed to upload file"}), 500
            
    except Exception as e:
        return jsonify({"error": str(e)}), 500

if __name__ == "__main__":
    app.run(debug=True)


   

File: uploadFileS3.js
// Import AWS SDK v3
const { S3Client, PutObjectCommand } = require('@aws-sdk/client-s3');
const fs = require('fs');
const path = require('path');
require('dotenv').config();

// Configure AWS
const s3Client = new S3Client({
    region: process.env.S3_REGION || 'us-east-1',
    credentials: {
        accessKeyId: process.env.AWS_ACCESS_KEY_ID,
        secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY
    }
});

/**
 * Uploads a file to S3
 * @param {string} filePath - Path to the file to upload
 * @param {string} bucketName - S3 bucket name
 * @param {string} [contentType='application/octet-stream'] - Content type of the file
 * @returns {Promise<string>} - URL of the uploaded file
 */
async function uploadFileToS3(filePath, bucketName, contentType = 'application/octet-stream') {
    try {
        // Get the filename from the path
        const fileName = path.basename(filePath);
        console.log('Uploading file:', fileName);
        console.log('Content type:', contentType);

        // Read the file
        const fileStream = fs.createReadStream(filePath);

        // Upload parameters
        const command = new PutObjectCommand({
            Bucket: bucketName,
            Key: fileName,
            Body: fileStream,
            ContentType: contentType,
            ACL: 'public-read'
        });

        // Upload to S3
        await s3Client.send(command);
        
        // Construct the URL
        const fileUrl = `https://${bucketName}.s3.${process.env.S3_REGION || 'us-east-1'}.amazonaws.com/${fileName}`;
        
        console.log('File uploaded successfully!');
        console.log('File URL:', fileUrl);
        return fileUrl;
    } catch (error) {
        console.error('An error occurred while uploading to S3:');
        console.error(error);
        return null;
    }
}

/**
 * Main function to handle file upload
 */
async function main() {
    const filePath = 'C:/Code/UploadFile/file.txt';
    const bucketName = process.env.S3_BUCKET_NAME;

    if (!bucketName) {
        console.error('S3_BUCKET_NAME environment variable is not set');
        return;
    }

    try {
        const fileUrl = await uploadFileToS3(filePath, bucketName);
        if (fileUrl) {
            console.log('File uploaded successfully!');
            console.log('File URL:', fileUrl);
        } else {
            console.log('File upload failed!');
        }
    } catch (error) {
        console.error('Error in main function:', error);
    }
}

// Run the main function if this file is executed directly
if (require.main === module) {
    main();
}

// Export the upload function for use in other modules
module.exports = {
    uploadFileToS3
}; 